src/QAT/qat/purr/integrations/tket.py

```diff
@@ -366,37 +389,33 @@ def run_tket_optimizations(qasm_string, opts, hardware: QuantumHardwareModel) ->
         )
         return qasm_string
 
-    couplings = hardware.qubit_direction_couplings
+    couplings = deepcopy(hardware.qubit_direction_couplings)
     optimizations_failed = False
     architecture = None
     if any(couplings):
         # Without default remapping pass multi-qubit gates don't get moved around, so
         # trying to apply them to a limited subset of qubits provides no value.
+        logical_qubit_map = {q.index: i for i, q in enumerate(hardware.qubits)}
```

Comment:
this is a property on the HW already. We don't want to be redefining this property any more than once, that'll let the demons in

Reply:
No it's not.

Reply:
ah yeah, fair point, that's an internal qat thing currently. Can we get a TODO here just to remember to consolidate once we go back over all this stiff